transport/graphsync/graphsync_test.go

```diff
@@ -1338,21 +1338,31 @@ func assertDecodesToMessage(t *testing.T, data datamodel.Node, expected datatran
 func assertHasOutgoingMessage(t *testing.T, extensions []graphsync.ExtensionData, expected datatransfer.Message) {
 	nd, err := expected.ToIPLD()
 	require.NoError(t, err)
-	expectedExt := graphsync.ExtensionData{
-		Name: extension.ExtensionDataTransfer1_1,
-		Data: nd,
+	found := false
+	for _, e := range extensions {
+		if e.Name == extension.ExtensionDataTransfer1_1 {
+			require.True(t, ipld.DeepEqual(nd, e.Data), "data matches")
```

Comment:
This is cause e.Data is a generic basicnode.Any node while nd is a representation of the message node format?

Reply:
also I'm glad ipld.DeepEqual exists :)

Reply:
Yeah, I think this is because the data doesn't do a full round-trip through binary so you end up with a typed version which fails a normal match, or something like that. It was just easier to do a DeepEqual than try and make the types perfectly the same. But we could also force a round-trip somewhere in here.

I'm reconsidering some of the serialisation stuff in the next PR, maybe these things will change a little by the time I get back to here.